pkg/okteto/pipeline_test.go

```diff
@@ -14,9 +14,760 @@
+					"deployment/queued":       "queued",
+					"deployment/progressing":  "progressing",
+					"statefulset/error":       "error",
+					"statefulset/queued":      "queued",
+					"statefulset/progressing": "progressing",
+					"job/error":               "error",
+					"job/queued":              "queued",
+					"job/progressing":         "progressing",
+					"cronjob/error":           "error",
+					"cronjob/queued":          "queued",
+					"cronjob/progressing":     "progressing",
+				},
+				err: nil,
+			},
+		},
+		{
+			name: "no error - devenv only deploy errors",
+			input: input{
+				client: &fakeGraphQLClient{
+					queryResult: &getPipelineResources{
+						Response: previewResourcesStatus{
+							Deployments: []resourceInfo{
+								{
+									ID:         "test",
+									Name:       "error",
+									Status:     "error",
+									DeployedBy: "1",
+								},
+								{
+									ID:         "test",
+									Name:       "queued",
+									Status:     "queued",
+									DeployedBy: "2",
+								},
+								{
+									ID:         "test",
+									Name:       "progressing",
+									Status:     "progressing",
+									DeployedBy: "3",
+								},
+							},
+							Statefulsets: []resourceInfo{
+								{
+									ID:         "test",
+									Name:       "error",
+									Status:     "error",
+									DeployedBy: "1",
+								},
+								{
+									ID:         "test",
+									Name:       "queued",
+									Status:     "queued",
+									DeployedBy: "2",
+								},
+								{
+									ID:         "test",
+									Name:       "progressing",
+									Status:     "progressing",
+									DeployedBy: "3",
+								},
+							},
+							Jobs: []resourceInfo{
+								{
+									ID:         "test",
+									Name:       "error",
+									Status:     "error",
+									DeployedBy: "1",
+								},
+								{
+									ID:         "test",
+									Name:       "queued",
+									Status:     "queued",
+									DeployedBy: "2",
+								},
+								{
+									ID:         "test",
+									Name:       "progressing",
+									Status:     "progressing",
+									DeployedBy: "3",
+								},
+							},
+							Cronjobs: []resourceInfo{
+								{
+									ID:         "test",
+									Name:       "error",
+									Status:     "error",
+									DeployedBy: "1",
+								},
+								{
+									ID:         "test",
+									Name:       "queued",
+									Status:     "queued",
+									DeployedBy: "2",
+								},
+								{
+									ID:         "test",
+									Name:       "progressing",
+									Status:     "progressing",
+									DeployedBy: "3",
+								},
+							},
+						},
+					},
+					err: nil,
+				},
+				namespace:  "test",
+				devenvName: "1",
+			},
+			expected: expected{
+				response: map[string]string{
+					"deployment/error":  "error",
+					"statefulset/error": "error",
+					"job/error":         "error",
+					"cronjob/error":     "error",
+				},
+				err: nil,
+			},
+		},
```

Comment:
Should we add a check for the not found?

```go
{
    name: "not found",
    input: input{
        client: &fakeGraphQLClient{
            queryResult: nil,
            err:         oktetoErrors.ErrNotFound,
        },
        namespace:  "test",
        devenvName: "non-existing",
    },
    expected: expected{
        response: nil,
        err:      fmt.Errorf("could not create okteto client"),
    },
},
```

Altough I think this will fail to compare the error (I am curious if you know how to fix it?)

Also, we could add a test for the retry mechanism (for any non-notfound errors but I guess we would need to extend the client mock to return once a failure and next a good result, so maybe it's not worth it now)

Reply:
Done. The way to do it is extract the error to a variable containing the message. That way whenever we compare the error with `errors.Is` we'll be pointing to the same error and we won't need to use the string comparison